engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala

```diff
@@ -45,11 +45,7 @@ trait TypeMatchers {
       expr: IR.Expression
     ): Option[(Sig, IR.Expression, String)] = (sig, expr) match {
       case (Name(n), t: IR.Name.Literal) if n == t.name =>
-        if (n == t.name) {
```

Comment:
This is real logic, making sure we verify names correctly

Reply:
Yes, but it is already guarded by the condition one line above

Reply:
Bringing back custom error, as discussed